test/functional/specs/desktop/C2581.js

```diff
@@ -22,20 +23,43 @@ test.meta({
   TEST_RUN: "Regression"
 });
 
+const triggerAlloyEvents = ClientFunction(() => {
+  window.alloy("event", { viewStart: true, data: { key: "value" } });
+  window.alloy("event", { data: { key: "value" } });
+  window.alloy("event", { data: { key: "value" } });
+});
+
+const identityCookieName = "kndctr_53A16ACB5CC1D3760A495C99_AdobeOrg_identity";
+
 test("Test C2581: Queue requests until we receive an ECID.", async () => {
-  await t.click(Selector("#debugEnabled-button"));
-  await t.click(Selector("#event-button"));
-  await t.click(Selector("#event-button"));
-  await t.click(Selector("#event-button"));
+  await configureAlloyInstance("alloy", debugEnabledConfig);
 
-  await responseStatus(networkLogger.edgeEndpointLogs.requests, 200);
+  await triggerAlloyEvents();
 
   await t.expect(networkLogger.edgeEndpointLogs.requests.length).eql(1);
 
-  const ecidResponse = getResponseBody(
+  await t.wait(1000);
+
+  const responseBody = getResponseBody(
     networkLogger.edgeEndpointLogs.requests[0]
   );
+  const response = networkLogger.edgeEndpointLogs.requests[0].response;
+  const responseHeader = response.headers;
+
+  const stringifiedResponseBody = JSON.stringify(responseBody);
+
+  const hasIdentityCookieInBody = stringifiedResponseBody.includes(
+    identityCookieName
+  );
```

Comment:
Do you know why we stringify the response body and then search the result rather than just looking in the `responseBody` object?

Reply:
Not really, it should be looked up in the object, I kept the logic as is and refactored only. I'm gonna make that change though, thank you!